### ISSUE TITLE
Counterfeiter should generate fakes for interfaces whose package name does not match directory name

### ISSUE THREAD

**Issue by challiwill** [maxbrunsfeld/counterfeiter#54]
I have an interface:

``` go
import "google.golang.org/api/storage/v1"

type StorageService interface {
    Buckets(string) (*storage.Buckets, error)
}
```

counterfeiter panics with 

```
$ counterfeiter . StorageService
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1239a8]

goroutine 1 [running]:
panic(0x199d00, 0xc42000e0d0)
    /usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:500 +0x1a1
github.com/maxbrunsfeld/counterfeiter/astutil.InjectAlias.func1(0x406420, 0xc4200a6a20, 0xc41fffa401)
    /Users/pivotal/go/src/github.com/maxbrunsfeld/counterfeiter/astutil/mutator.go:11 +0x988
go/ast.inspector.Visit(0xc42023cf60, 0x406420, 0xc4200a6a20, 0x2f56e0, 0xc42023cf60)
    /usr/local/Cellar/go/1.7.1/libexec/src/go/ast/walk.go:373 +0x3a
go/ast.Walk(0x2f56e0, 0xc42023cf60, 0x406420, 0xc4200a6a20)
    /usr/local/Cellar/go/1.7.1/libexec/src/go/ast/walk.go:52 +0x63
go/ast.Walk(0x2f56e0, 0xc42023cf60, 0x2f5e20, 0xc42009ab80)
    /usr/local/Cellar/go/1.7.1/libexec/src/go/ast/walk.go:74 +0xd22
go/ast.Walk(0x2f56e0, 0xc42023cf60, 0x2f5e60, 0xc42007ecf0)
    /usr/local/Cellar/go/1.7.1/libexec/src/go/ast/walk.go:84 +0x152
go/ast.Walk(0x2f56e0, 0xc42023cf60, 0x2f5ee0, 0xc4200a6aa0)
    /usr/local/Cellar/go/1.7.1/libexec/src/go/ast/walk.go:168 +0x22df
go/ast.Inspect(0x2f5ee0, 0xc4200a6aa0, 0xc42023cf60)
    /usr/local/Cellar/go/1.7.1/libexec/src/go/ast/walk.go:385 +0x4b
github.com/maxbrunsfeld/counterfeiter/astutil.InjectAlias(0xc4200a6aa0, 0xc42007fe90, 0xc42007ff20)
    /Users/pivotal/go/src/github.com/maxbrunsfeld/counterfeiter/astutil/mutator.go:23 +0x92
github.com/maxbrunsfeld/counterfeiter/generator.CodeGenerator.fixup(0x7fff5fbff788, 0xe, 0xc42007b760, 0x1, 0x1, 0xc42009a816, 0x27, 0xc42007a750, 0x3, 0x1, ...)
    /Users/pivotal/go/src/github.com/maxbrunsfeld/counterfeiter/generator/generator.go:191 +0x80
github.com/maxbrunsfeld/counterfeiter/generator.CodeGenerator.buildASTForFake(0x7fff5fbff788, 0xe, 0xc42007b760, 0x1, 0x1, 0xc42009a816, 0x27, 0xc42007a750, 0x3, 0x1, ...)
    /Users/pivotal/go/src/github.com/maxbrunsfeld/counterfeiter/generator/generator.go:61 +0x1a3
github.com/maxbrunsfeld/counterfeiter/generator.CodeGenerator.GenerateFake(0x7fff5fbff788, 0xe, 0xc42007b760, 0x1, 0x1, 0xc42009a816, 0x27, 0xc42007a750, 0x3, 0x1, ...)
    /Users/pivotal/go/src/github.com/maxbrunsfeld/counterfeiter/generator/generator.go:29 +0xc2
main.main()
    /Users/pivotal/go/src/github.com/maxbrunsfeld/counterfeiter/main.go:57 +0x3f9
```

The issue is fixed when I alias the import:

``` go
import storage "google.golang.org/api/storage/v1"

type StorageService interface {
    Buckets(string) (*storage.Buckets, error)
}
```

**Comment by rishabhtulsian** [comment]
+1

**Comment by joefitzgerald** [comment]
This is fixed in v6.0.0.
